File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_set>
using namespace std;

int solution(vector<int> nums)
{
    unordered_set<int> us;
    for (vector<int>::iterator it = nums.begin(); it != nums.end(); it++)
        us.insert(*it);
    
    return us.size() < nums.size() / 2 ? us.size() : nums.size() / 2;
}

File: Java/프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.Set;
import java.util.HashSet;


class Solution {
    public int solution(int[] nums) {
        Set<Integer> option = new HashSet<>();    
        for (int n : nums)
            option.add(n);
        int unique = option.size();
        int choice = nums.length / 2;
        return Math.min(unique, choice);
    }
}